src/util/irep.h

```diff
@@ -82,8 +82,77 @@ inline const std::string &name2string(const irep_namet &n)
 class irept;
 const irept &get_nil_irep();
 
-/*! \brief Base class for tree-like data structures with sharing
-*/
+/// \brief Base class for tree-like data structures with sharing
+///
+/// There are a large number of kinds of tree structured or tree-like data in
+/// CPROVER. `irept` provides a single, unified representation for all of
+/// these, allowing structure sharing and reference counting of data. As
+/// such `irept` is the basic unit of data in CPROVER.  Each `irept`
+/// contains[^1] a basic unit of data (of type `dt`) which contains four
+/// things:
+///
+/// * `data`:   A string[^2], which is returned when the `id()` function is
+///   used.
+///
+/// * `named_sub`:   A map from `irep_namet` (a string) to `irept`. This
```

Comment:
Use as many \ref as possible

Reply:
Annoyingly, doxygen sometimes can't get the \ref to work if the name is in backticks, but sometimes it can